components/places/src/api/matcher.rs

```diff
@@ -78,7 +79,17 @@ pub fn search_frecent(conn: &PlacesDb, params: SearchParams) -> Result<Vec<Searc
         params.limit,
     )?;
 
-    Ok(matches)
+    // dedup matches by their URLs
+    let mut unique_matches: Vec<SearchResult> = Vec::new();
```

Comment:
Hm, It's not ideal that we have to do a clone of every item here.

I think you could avoid this by using [`sort_unstable_by_key`](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.sort_unstable_by_key) followed by [`dedup_by_key`](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.dedup_by_key). (Using `search_result.url` for both).